api/numaresourcesoperator/v1/numaresourcesoperator_types.go

```diff
@@ -125,6 +125,28 @@ type NodeGroup struct {
 	PoolName *string `json:"poolName,omitempty"`
 }
 
+// NodeGroupStatus reports the status of a NodeGroup once matches an actual set of nodes and it is correctly processed
+// by the system. In other words, is not possible to have a NodeGroupStatus which does not represent a valid NodeGroup
+// which in turn correctly references unambiguously a set of nodes in the cluster.
+// Hence, if a NodeGroupStatus is published, its `Name` must be present, because it refers back to a NodeGroup whose
+// config was correctly processed in the Spec. And its DaemonSet will be nonempty, because matches correctly a set
+// of nodes in the cluster. The Config is best-effort always represented, possibly reflecting the system defaults.
+// If the system cannot process a NodeGroup correctly from the Spec, it will report Degraded state in the top-level
+// condition, and will provide details using the aforementioned conditions.
+type NodeGroupStatus struct {
+	// DaemonSet of the configured RTEs, for this node group
+	// +optional
+	//+operator-sdk:csv:customresourcedefinitions:type=status,displayName="RTE DaemonSets"
+	DaemonSet *NamespacedName `json:"daemonsets,omitempty"`
+	// NodeGroupConfig represents the latest available configuration applied to this NodeGroup
+	// +optional
+	//+operator-sdk:csv:customresourcedefinitions:type=status,displayName="Optional configuration enforced on this NodeGroup"
+	Config *NodeGroupConfig `json:"config,omitempty"`
```

Comment:
Likewise, every `NodeGroup` managed by the Operator because it is reconciling a `NUMAResourcesOperator` object must have a `NodeGroupConfig`. If not supplied by the user, we are using defaults, which we must publish.
This field is thus required, not optional, thus it should _not_ be a pointer _nor_ it should be `omitempty`.

Reply:
makes sense, adjusted in latest updates and updated controller logic when publishing NodeGroupStatus. Thanks!